src/inmanta/protocol/rest/__init__.py

```diff
@@ -189,6 +189,38 @@ async def _run_getters(self, arg: str, value: Optional[object]) -> Optional[obje
             LOGGER.exception("Failed to use getter for arg %s", arg)
             raise e
 
+    @staticmethod
+    def _handle_get_request_arg(arg_type: Optional[Type[object]], message_arg: object) -> object:
+        """
+        Handles processing of arguments for GET requests, especially for list types encoded as URL query parameters.
+        Maps single element lists provided as query parameters to actual lists.
+        """
+        if typing_inspect.is_optional_type(arg_type):
+            non_none_arg_types = [arg for arg in typing_inspect.get_args(arg_type) if arg is not type(None)]
+            if len(non_none_arg_types) == 1 and typing_inspect.get_origin(non_none_arg_types[0]) is list:
+                arg_type = non_none_arg_types[0]
+
+        is_generic_list = arg_type and typing_inspect.is_generic_type(arg_type) and typing_inspect.get_origin(arg_type) is list
+        is_single_element_list = len(typing_inspect.get_args(arg_type, evaluate=True)) == 1
+        is_not_list = not isinstance(message_arg, list)
+
+        if is_generic_list and is_not_list and is_single_element_list:
+            return [message_arg]
+        return message_arg
+
+    def _validate_argument_consistency(self, args):
+        """
+        Validates the consistency of arguments, ensuring they are not passed both as a header and a non-header value.
+        """
+        for arg in args:
+            if arg in self._message and self._is_header_param(arg) and self._is_header_param_provided(arg):
+                message_value = self._message[arg]
+                header_value = self._get_header_value_for(arg)
+                if message_value != header_value:
+                    raise exceptions.BadRequest(
+                        f"Argument {arg} has inconsistent values (header={header_value}; non-header={message_value})."
```

Comment:
Why was this error message changed? The previous message is more clear in my opinion?